apps/src/blockly/addons/cdoXml.js

```diff
@@ -74,6 +75,50 @@ export default function initializeBlocklyXml(blocklyWrapper) {
   blocklyWrapper.Xml.createBlockOrderMap = createBlockOrderMap;
 }
 
+/**
+ * Adds a mutation element to a block if it should have an open miniflyout.
+ * CDO Blockly uses an unsupported method for serializing miniflyout state
+ * where arbitrary block attribute could be used to manage extra state.
+ * Mainline Blockly expects a mutator. The presence of the mutation element
+ * will trigger the block's domToMutation function to run, if it exists.
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ */
+export function addMutationToMiniToolboxBlocks(blockElement) {
+  const miniflyoutAttribute = blockElement.getAttribute('miniflyout');
+  if (!miniflyoutAttribute) {
+    return; // Not a miniflyout block
+  }
+  // The default icon is a '+' symbol which represents a currently-closed flyout.
+  const useDefaultIcon = miniflyoutAttribute === 'open' ? 'false' : 'true';
+  // We don't expect a mutation to exist yet, but check just in case.
+  const existingMutationElement = blockElement.querySelector('mutation');
+
+  if (!existingMutationElement) {
+    // The mutation element does not exist, so create it.
+    const newMutationElement =
+      blockElement.ownerDocument.createElement('mutation');
+
+    // Create new mutation attribute based on original block attribute.
+    newMutationElement.setAttribute('useDefaultIcon', useDefaultIcon);
+
+    // Place mutator before fields, values, and other nested blocks.
+    blockElement.insertBefore(newMutationElement, blockElement.firstChild);
+  } else {
+    // The mutation element already exists, inspect it.
+    const existingUseDefaultIcon =
```

Comment:
Do we need to handle this case? Or can we assume that if we have an existing mutation, it was added correctly?

Reply:
I don't think we need to handle this case since this is the only way the mutation (and this attribute) can be added. In fact, I'd argue this step might be overkill... If there's already a mutation, we can realistically assume that the attribute is there and set correctly.

Reply:
Specifically, I'm wondering if this frees us up to simplify this by doing something like...

```
const miniflyoutAttribute = blockElement.getAttribute('miniflyout');
const existingMutationElement = blockElement.querySelector('mutation');
if (!miniflyoutAttribute || existingMutationElement) {
  return;
}

// other stuff

```

Reply:
Oh that makes sense! I think that would work just fine.

Reply:
Updated!